application/src/main/java/org/togetherjava/tjbot/features/help/HelpSystemHelper.java

```diff
@@ -131,6 +144,94 @@ private RestAction<Message> sendExplanationMessage(GuildMessageChannel threadCha
+     *         why the message wasn't used.
+     */
+    RestAction<Message> constructChatGptAttempt(ThreadChannel threadChannel,
+            String originalQuestion) {
+
+        Optional<String> questionOptional = prepareChatGptQuestion(threadChannel, originalQuestion);
+        Optional<String> chatGPTAnswer;
+
+        String question = questionOptional.orElseThrow();
+        logger.debug("The final question sent to chatGPT: {}", question);
+        chatGPTAnswer = chatGptService.ask(question);
+
+        if (chatGPTAnswer.isEmpty()) {
+            logger.warn("Something went wrong while trying to communicate with the ChatGpt API");
+            return useChatGptFallbackMessage(threadChannel);
+        }
+
+        String aiResponse = chatGPTAnswer.get();
+        String lowercaseAiResponse = aiResponse.toLowerCase(Locale.US);
+        if (lowercaseAiResponse.contains("as an ai language model")) {
+            logger.debug("Response from ChatGPT, lacks context or unable to respond?: {}",
+                    aiResponse);
+            return useChatGptFallbackMessage(threadChannel);
+        }
+
+        UnaryOperator<String> response = """
+                Here is an AI assisted attempt to answer your question 🤖. Maybe it helps! \
+                In any case, a human is on the way 👍. To continue talking to the AI, you can use \
+                %s.
+                """::formatted;
+
+        return mentionGuildSlashCommand(threadChannel.getGuild(), ChatGptCommand.COMMAND_NAME)
+            .map(response)
+            .flatMap(threadChannel::sendMessage)
+            .flatMap(embed -> threadChannel
+                .sendMessageEmbeds(HelpSystemHelper.embedWith(aiResponse)));
+    }
+
+    private Optional<String> prepareChatGptQuestion(ThreadChannel threadChannel,
+            String originalQuestion) {
+        String questionTitle = threadChannel.getName();
+        StringBuilder questionBuilder = new StringBuilder(MAX_QUESTION_LENGTH);
+
+        if (originalQuestion.length() < MIN_QUESTION_LENGTH
+                && questionTitle.length() < MIN_QUESTION_LENGTH) {
+            return Optional.empty();
+        }
+
+        questionBuilder.append(questionTitle).append(" ");
+        if (originalQuestion.length() > MAX_QUESTION_LENGTH - questionBuilder.length()) {
+            originalQuestion =
+                    originalQuestion.substring(0, MAX_QUESTION_LENGTH - questionBuilder.length());
+        }
+
+        questionBuilder.append(originalQuestion);
+
+        StringBuilder tagBuilder = new StringBuilder();
+        int stringLength = questionBuilder.length();
+        for (ForumTag tag : threadChannel.getAppliedTags()) {
+            String tagName = tag.getName();
+            stringLength += tagName.length();
+            if (stringLength > MAX_QUESTION_LENGTH) {
+                break;
+            }
+            tagBuilder.append(String.format("%s ", tagName));
+        }
```

Comment:
replace with a stream?

```suggestion
        String tagsJoined = threadChannel.getAppliedTags().stream()
           .map(ForumTag::getName)
           .filter(tag -> tag.length() > MAX_QUESTION_LENGTH)
           .collect(Collectors.joining(" "));
```